From 719b58c1fdd69149033cdd8d31f5adbb920eefa1 Mon Sep 17 00:00:00 2001
From: Kuppuswamy Sathyanarayanan <sathyanarayanan.kuppuswamy@linux.intel.com>
Date: Mon, 21 May 2018 14:54:05 -0700
Subject: [PATCH 1014/1240] Revert "extcon: USB: Use static DRD mode for
 host/device mode switch"

This reverts commit 2ea804f5435ef30d534295a0fd9faf693fbeab64.

Signed-off-by: Kuppuswamy Sathyanarayanan <sathyanarayanan.kuppuswamy@linux.intel.com>
---
 drivers/extcon/extcon-intel-usb.c | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/drivers/extcon/extcon-intel-usb.c b/drivers/extcon/extcon-intel-usb.c
index 26b043d..9290eb7 100644
--- a/drivers/extcon/extcon-intel-usb.c
+++ b/drivers/extcon/extcon-intel-usb.c
@@ -49,10 +49,10 @@ static int intel_usb_mux_notifier(struct notifier_block *nb,
 	u32 val;
 
 	if (mux->edev.state)
-		val = CFG0_SW_IDPIN_EN | CFG0_SW_DRD_STATIC_HOST | CFG0_SW_SWITCH_EN;
+		val = CFG0_SW_IDPIN_EN | CFG0_SW_DRD_STATIC_HOST;
 	else
 		val = CFG0_SW_IDPIN_EN | CFG0_SW_IDPIN | CFG0_SW_VBUS_VALID |
-		      CFG0_SW_DRD_STATIC_DEV | CFG0_SW_SWITCH_EN ;
+		      CFG0_SW_DRD_STATIC_DEV;
 
 	writel(val, mux->regs);
 	return NOTIFY_OK;
@@ -63,7 +63,6 @@ struct intel_usb_mux *intel_usb_mux_register(struct device *dev,
 {
 	struct intel_usb_mux *mux;
 	int ret;
-	u32 val;
 
 	mux = kzalloc(sizeof(*mux), GFP_KERNEL);
 	if (!mux)
@@ -75,10 +74,6 @@ struct intel_usb_mux *intel_usb_mux_register(struct device *dev,
 		return ERR_PTR(-ENOMEM);
 	}
 
-	val = CFG0_SW_IDPIN_EN | CFG0_SW_IDPIN | CFG0_SW_VBUS_VALID |
-			CFG0_SW_DRD_STATIC_DEV | CFG0_SW_SWITCH_EN;
-	writel(val, mux->regs);
-
 	mux->edev.dev.parent = dev;
 	mux->edev.supported_cable = intel_mux_cable;
 
-- 
2.7.4

